### ISSUE TITLE
Able to install ingress-nginx chart using helm cli, but not with the helm go package v3.6.0

### ISSUE THREAD

**Issue by sebastianmontero** [helm/helm#9772]
I'm able to install the ingress-nginx chart version 3.32.0 from repo https://kubernetes.github.io/ingress-nginx without any issues using the helm cli, but when installing it using the helm go package it times out waiting for the job ingress-nginx-admission-create to complete.  Why might this be happening?

Output of `helm version`: v3.6.0

helm go package version: v3.6.0

Output of `kubectl version`: 

Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.5", GitCommit:"6b1d87acf3c8253c123756b9e61dac642678305f", GitTreeState:"clean", BuildDate:"2021-03-18T01:10:43Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"linux/amd64"}

Server Version: version.Info{Major:"1", Minor:"19+", GitVersion:"v1.19.9-gke.1400", GitCommit:"ec68c7064ea987ad0c7fb63930df96bdefeb93c4", GitTreeState:"clean", BuildDate:"2021-04-07T09:20:04Z", GoVersion:"go1.15.8b5", Compiler:"gc", Platform:"linux/amd64"}

Cloud Provider/Platform (AKS, GKE, Minikube etc.): GKE

**Comment by mattfarina** [comment]
The Helm CLI uses the Go SDK so this is curious. How are you using the Go package? Is there anything you can share, for example how you're using the API or any logs?

**Comment by sebastianmontero** [comment]
Hi @mattfarina 
I've developed a basic exmaple:

``` package main

import (
	"fmt"
	"log"
	"os"
	"path"
	"time"

	"helm.sh/helm/v3/pkg/action"
	"helm.sh/helm/v3/pkg/chart/loader"
	"helm.sh/helm/v3/pkg/cli"
	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
)

func main() {
	namespace := "ingress-basic"
	chartName := "ingress-nginx"
	chartDir := "/home/sebastian/helm-chart-repo"
	repoURL := "https://kubernetes.github.io/ingress-nginx"
	version := "3.32.0"
	releaseName := "test-release"

	settings := cli.New()
	settings.Debug = true
	actionConfig := new(action.Configuration)
	// You can pass an empty string instead of settings.Namespace() to list
	// all namespaces
	if err := actionConfig.Init(settings.RESTClientGetter(), namespace, os.Getenv("HELM_DRIVER"), log.Printf); err != nil {
		log.Panicf("failed to create action config, error: %v", err)
	}

	fmt.Printf("Pulling chart: %v from repo: %v\n", chartName, repoURL)
	pullClient := action.NewPullWithOpts(action.WithConfig(actionConfig))
	pullClient.RepoURL = repoURL
	pullClient.Settings = settings
	pullClient.DestDir = chartDir
	pullClient.Version = version
	_, err := pullClient.Run(chartName)
	if err != nil {
		log.Panicf("failed to pull chart: %v, error: %v", chartName, err)
	}

	client := action.NewInstall(actionConfig)
	client.Namespace = namespace
	client.ReleaseName = releaseName
	client.Wait = true
	client.WaitForJobs = true
	client.CreateNamespace = true
	client.Timeout = time.Minute * 3

	chartPath := path.Join(chartDir, fmt.Sprintf("%v-%v.tgz", chartName, version))
	chart, err := loader.Load(chartPath)
	if err != nil {
		log.Panicf("failed to load chart: %v\nerror: %v", chartPath, err)
	}
	values := map[string]interface{}{
		"controller": map[string]interface{}{
			"replicaCount": 2,
		},
	}
	fmt.Printf("Installing chart: %v, releaseName: %v\n", chartName, releaseName)
	release, err := client.Run(chart, values)
	if err != nil {
		log.Panicf("failed installing chart, error: %v", err)
	}
	fmt.Printf("Chart: %v installed", release)
}
 ```

This is the output I get: 
![image](https://user-images.githubusercontent.com/13155714/120809546-cbd75400-c50f-11eb-9c03-5afda4afbfb7.png)

**Comment by bacongobbler** [comment]
I noticed that you set the “wait” flag to true. Can you check and confirm whether you do the same with the helm CLI, and if so, check the logs of the pre-install hook that is causing the failure?

**Comment by sebastianmontero** [comment]
> I noticed that you set the “wait” flag to true. Can you check and confirm whether you do the same with the helm CLI, and if so, check the logs of the pre-install hook that is causing the failure?

With the cli I have installed it with or without the wait flag and both work without any issues. And I've ran the example with the flags on and off, it fails in both cases. 

Where can I check the hook logs?

**Comment by bacongobbler** [comment]
`kubectl logs`

**Comment by bacongobbler** [comment]
potentially relevant: https://github.com/helm/helm/issues/9761

**Comment by sebastianmontero** [comment]
> `kubectl logs`

The log for the job only contains:
"error: timed out waiting for the condition"
How can I get the logs for the preinstall web hook

Here is the describe for the job:
![image](https://user-images.githubusercontent.com/13155714/120824017-a0f3fc80-c51d-11eb-9294-70b5f9fc5338.png)

**Comment by sebastianmontero** [comment]
> potentially relevant: #9761

I had seen that issue, but I am specifying a timeout.

**Comment by bacongobbler** [comment]
The pre-install web hook IS the job. So running `kubectl logs` on the job is the correct action here.

In this particular case, it would appear that your job has his a "backoff limit". A backoff limit tells your Job to stop trying after a number of failed attempts. This parameter is described [in the Kubernetes documentation](https://kubernetes.io/docs/concepts/workloads/controllers/job/).

From https://helm.sh/docs/topics/charts_hooks/#hooks-and-the-release-lifecycle:

> What does it mean to wait until a hook is ready? This depends on the resource declared in the hook. If the resource is a Job or Pod kind, Helm will wait until it successfully runs to completion. And if the hook fails, the release will fail. This is a blocking operation, so the Helm client will pause while the Job is run.

In other words, because your hook was declared as a Job, and the hook failed due to reaching the backoff limit, the release failed to install.

**Comment by bacongobbler** [comment]
Short answer: fix your Job, and the release should succeed.

**Comment by sebastianmontero** [comment]
> Short answer: fix your Job, and the release should succeed.

So, do you mean that increasing the timeout should fix the issue? I've increased it to 10 minutes and still fails, with the cli the installation does not take more than 2 minutes

**Comment by bacongobbler** [comment]
I'm saying that Kubernetes is informing you that one pod (the pre-install hook) is failing, causing the installation to fail. I would investigate further as to why that happened.

**Comment by sebastianmontero** [comment]
> Kubernetes is informing you that one pod (the pre-install hook) has failed, causing the installation to fail. I would investigate further as to why that happened. Increasing the timeout will not fix the issue.

But what would be the reason that the job does not fail when I install the chart using the helm cli, but fails when I install it using the helm go package? I did not develop this chart.

**Comment by bacongobbler** [comment]
It's unclear to me at this time. Perhaps the logs from the failed job would tell us more.

> The log for the job only contains:
> "error: timed out waiting for the condition"

Can you share how you found that error message? Was that from `kubectl logs test-release-ingress-nginx-admission-create`?

**Comment by sebastianmontero** [comment]
> It's unclear to me at this time. Perhaps the logs from the failed job would tell us more.
> 
> > The log for the job only contains:
> > "error: timed out waiting for the condition"
> 
> Can you share how you found that error message? Was that from `kubectl logs test-release-ingress-nginx-admission-create`?

kubectl logs job/test-release-ingress-nginx-admission-create
